initalize logging after connection

antholeole opened the pull request:
after a connection is established, send a setLoggingLevel request

#610 

## Motivation and Context
<!-- Why is this change needed? What problem does it solve? -->

## How Has This Been Tested?
<!-- Have you tested this in a real application? Which scenarios were tested? -->

## Breaking Changes
None

## Types of changes
<!-- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update

## Checklist
<!-- Go over all the following points, and put an `x` in all the boxes that apply. -->
- [X] I have read the [MCP Documentation](https://modelcontextprotocol.io)
- [ ] My code follows the repository's style guidelines
- [ ] New and existing tests pass locally
- [ ] I have added appropriate error handling
- [ ] I have added or updated documentation as needed

olaservo commented on client/src/App.tsx:
```diff
@@ -204,6 +204,7 @@ const App = () => {
       ]);
     },
     getRoots: () => rootsRef.current,
+    defaultLoggingLevel: "debug"
```
Should this just be: `defaultLoggingLevel: logLevel` since its already has the same default set in `useState<LoggingLevel>("debug")` ?

olaservo reviewed (changes requested):
I think it makes sense to add the `setLoggingLevel` to match the default selected level.  I just had one comment.  (And I think you have some linting errors that are failing the CI.)

Side note, I think the title of the issue you linked to this PR doesn't seem accurate, since a server could still emit logs without this being set.

antholeole commented:
@olaservo I see you added the waiting for submitter tag - is there something else to do on my end? I thought I addressed your comment, lmk if there’s anything else you want from me :)

olaservo commented:
Hi @antholeole , I'm good to merge this but for some reason some new formatting issues are showing up in the CI checks - could you resolve that?  (If I make those changes it won't count my approval.) Thanks!

KKonstantinov commented:
Hello,

This seem to lead to an issue discussed in #699, especially the code in useConnection 

https://github.com/modelcontextprotocol/inspector/pull/611/files#diff-1308b8974395372ca4d89e566b653312adcb1a690a926bb21431639928db2701R566-R569

Please check?

KKonstantinov commented:
And here is what happens:

whenever logging/setLevel is called, the MCP server has no default handler for this request, therefore it goes into the "Method not found" flow.

To resolve it, the MCP server has to register a requestHandler for this method, however I would consider this worth reverting until a more graceful way to handle this is implemented.

<img width="1723" height="416" alt="image" src="https://github.com/user-attachments/assets/9fed7498-f8b5-4591-99ee-061b4b5d5ebe" />

cliffhall commented:
The inspector is operating in spec - if the server advertises support for logging, the Inspector sends a setLevel request, as it "MAY" do according to spec. The server is operating out of spec. It advertises support for logging but fails to implement it, i.e., handle the setLevel request. 

Other clients in the wild which follow spec will naturally do this and when it fails with them, it's more likely a real "getting in the way of work" problem.

If we revert the automatic setLevel request, I think we're encouraging server developers to continue doing the wrong thing. This is after all, a tool for developers to inspect their servers and determine if they're operating correctly. We could possibly detect this particular case and display a toast to that effect? Would that be reasonable?
